homeassistant/components/metoffice/sensor.py

```diff
@@ -95,15 +106,22 @@ async def async_setup_entry(
 class MetOfficeCurrentSensor(SensorEntity):
```

Comment:
Why aren't we inheriting from CoordinatorEntity?

Reply:
You are absolutely right, we should use it! And it will definitely make things much simpler.